application/comit_node/src/settings/mod.rs

```diff
@@ -5,76 +5,165 @@ use crate::seed::Seed;
+                port: 8000,
+            },
+            btsieve: Btsieve {
+                url,
+                bitcoin: PollParameters {
+                    poll_interval_secs: Duration::from_secs(300),
+                    network: "regtest".into(),
+                },
+                ethereum: PollParameters {
+                    poll_interval_secs: Duration::from_secs(20),
+                    network: "regtest".into(),
+                },
+            },
+            web_gui: Some(HttpSocket {
+                address: IpAddr::V4(Ipv4Addr::LOCALHOST),
+                port: 8080,
+            }),
+            log_levels: LogLevels {
+                comit_node: LevelFilter::Debug,
+            },
+        }
+    }
 }
 
 fn default_log() -> LevelFilter {
     LevelFilter::Debug
 }
 
-#[derive(Clone, Debug, Deserialize)]
+fn default_log_levels() -> LogLevels {
+    LogLevels {
+        comit_node: default_log(),
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct LogLevels {
+    #[serde(with = "self::serde_log", default = "default_log")]
+    pub comit_node: LevelFilter,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct Comit {
     pub secret_seed: Seed,
 }
 
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct Network {
     pub listen: Vec<Multiaddr>,
 }
 
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct HttpSocket {
     pub address: IpAddr,
     pub port: u16,
 }
 
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct Btsieve {
     #[serde(with = "url_serde")]
     pub url: url::Url,
     pub bitcoin: PollParameters,
     pub ethereum: PollParameters,
 }
 
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct PollParameters {
     #[serde(with = "self::serde_duration")]
     pub poll_interval_secs: Duration,
     pub network: String,
 }
 
 impl ComitNodeSettings {
-    pub fn create<D: AsRef<OsStr>, R: AsRef<OsStr>>(
-        default_config: D,
-        run_mode_config: R,
-    ) -> Result<Self, ConfigError> {
+    pub fn create_with_default(path: PathBuf, file_name: &str) -> Result<Self, ConfigError> {
+        if !path.exists() {
+            log::warn!(
+                "Config path does not exist, creating folders recursively: {:?}",
+                path
+            );
+            fs::create_dir_all(path.clone()).map_err(|error| {
+                ConfigError::Message(format!("Could not create folders: {:?}: {:?}", path, error))
+            })?;
+        }
+
+        let default_config = path.join(file_name);
+
+        if default_config.exists() {
+            if default_config.is_file() {
+                log::warn!(
+                    "Config files exists, loading config from: {:?}",
```

Comment:
📣  It means that the `create_with_default` was called but no creation is happening because a file already exist. I think we should error out here.

Reply:
Ok, can do so, any other opinions on this?

Reply:
Yeah I was thinking the same.

I might make sense to change the API to be:

```rs
fn write_to(&self, path: PathBuf);
```

and combine it with `Config::default()` with something like:

High-level flow:

1. If --config is passed, try to load it
2. If --config is not passed, try to load from default location
3. If not found, create new instance of config with `Config::default()` and write it to the default location with `config.write_to(DEFAULT_LOCATION)`.

Reply:
@thomaseizinger : that's what I am doing right now. Do think of controlling this from the outside though?

Reply:
updated